electron/golem_handler.js

```diff
@@ -17,29 +18,66 @@ class GolemProcess {
         this.processArgs = processArgs || ['--nogui', '-r', '127.0.0.1:61000'];
     }
 
-    startProcess(err, pid) {
-        if (!this.process)
+    startProcess() {
+        try {
+            if (this.process) return;
+            if (!this.checkVersion()) return;
             this._startProcess();
+        } catch (e) {
+            console.error('GolemProcess error:', e);
+        }
     }
 
-    _startProcess() {
+    prepareSpawnOptions() {
         let cwd = path.join(os.homedir(), '.golem');
-        let env = process.env;
-
         /* Create a working directory */
         if (!fs.existsSync(cwd))
             fs.mkdirSync(cwd);
 
+        let env = process.env;
         /* Patch PATH on Unix and Linux */
         if (os.platform() != 'win32')
             env.PATH += ':/usr/local/bin';
-
-        console.log('💻 Starting Golem...');
-        this.process = spawn(this.processName, this.processArgs, {
+        return {
             cwd: cwd,
             env: env,
-            stdio: 'ignore'
-        });
+        }
+    }
+
+    checkVersion() {
+        let options = this.prepareSpawnOptions();
+        options.stdio = 'pipe';
+        options.timeout = 1000; // ms
+        options.killSignal = 'SIGKILL';
+        let result = spawnSync(this.processName, ['--version'], options);
+        if (result.status != 0) {
+            console.error('Cannot start Golem process. Exit code:',
+                    result.status,
+                    'stderr:',
+                    result.stderr.toString()
+            );
+            return false;
+        }
+        let version_re = /^GOLEM version: (\d+\.\d+\.\d+)$/m;
+        let version_match = version_re.exec(result.stdout.toString());
+        let process_version = version_match[1].split(".");
```

Comment:
If someone doesn't have golemapp on the correct path, this line gives error like;
```
GolemProcess error: TypeError: Cannot read property '1' of null
at GolemProcess.checkVersion (/Users/mhmmd/projects/golem-electron/electron/golem_handler.js:63:44)
```

So I think we should handle this error, if path is not correct.